StRoot/StMuDSTMaker/COMMON/StMuDstMaker.cxx

```diff
@@ -1208,7 +1209,22 @@ void StMuDstMaker::fillFst(StEvent* ev) {
     connectFstCollection();
     mStMuDst->set(this);
   }
-  mFstUtil->fillMuFst(mFstCollection,fstcol);
+
+  //raw hit data input
+  StFstCollection* fstCollectionPtr = 0;
+  if (IAttr("fstMuRawHit")){//True for storing FST Raw hits
+    TObjectSet* fstDataSet = (TObjectSet*)GetDataSet("fstRawHitAndCluster");
+    if (!fstDataSet) {
+        LOG_WARN << "StMuDstMaker::Make() - there is no fstDataSet (raw hit) " << endm;
+    }        
+
+    fstCollectionPtr = (StFstCollection*)fstDataSet->GetObject();
```

Comment:
Although a warning is printed, a null `fstDataSet` pointer will be used here and cause possible problems. I think a more serious action is needed here to avoid using a null pointer. Perhaps return from the function if the `if (!fstDataSet)` clause, or replace this line with:

    fstCollectionPtr = (fstDataSet ? (StFstCollection*)fstDataSet->GetObject() : 0);
    
And then some similar logic should be done for when `fstCollectionPtr` is null.

Reply:
Thanks for the suggestion and I have done it for `fstDataSet`. For `fstCollectionPtr`, I have:
```
if(fstCollectionPtr) {
  ...
}
```
including all the cases the `fstCollectionPtr` is being used. So, I think it should be fine.